library/Haste/Ajax/ReloadHelper.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+namespace Haste\Ajax;
+
+use Haste\Http\Response\JsonResponse;
+
+class ReloadHelper
+{
+    const TYPE_CONTENT_ELEMENT = 'ce';
+    const TYPE_FRONTEND_MODULE = 'fmd';
+
+    /**
+     * Listeners
+     * @var array
+     */
+    private static $listeners = [];
+
+    /**
+     * Response
+     * @var array
+     */
+    private static $response = [];
+
+    /**
+     * Subscribe the listener
+     *
+     * @param string $type
+     * @param int    $id
+     * @param array  $events
+     */
+    public static function subscribe($type, $id, array $events)
+    {
+        $id = (int) $id;
+
+        foreach ($events as $event) {
+            if (!static::$listeners[$type][$event] || !in_array($id, static::$listeners[$type][$event], true)) {
+                static::$listeners[$type][$event][] = $id;
+            }
+        }
+    }
+
+    /**
+     * Store the response if applicable
+     *
+     * @param string $type
+     * @param int    $id
+     * @param string $event
+     * @param string $buffer
+     */
+    public static function storeResponse($type, $id, $event, $buffer)
+    {
+        if (!static::$listeners[$type][$event]) {
+            return;
+        }
+
+        $id = (int) $id;
+
+        foreach (static::$listeners[$type][$event] as $v) {
+            $key = static::getKey($type, $id);
+
+            if ($v !== $id || static::$response[$key]) {
+                continue;
+            }
+
+            static::$response[$key] = [
```

Comment:
do we need the value to be an array with ID again, if the ID is already the key?